horace_core/algorithms/cut.m

```diff
@@ -13,27 +13,41 @@
     % Get a loader instance that can tell us what kind of file this is
     % We expect either a .sqw or .dnd file, throw an error otherwise.
     ldr = sqw_formats_factory.instance().get_loader(source);
-    if ldr.sqw_type
-        % Load the .sqw file using the sqw constructor so that we can pass the
-        % pixel_page_size argument to get an sqw with file-backed pixels.
-        pixel_page_size = get(hor_config, 'pixel_page_size');
-        sqw_dnd_obj = sqw(source, 'pixel_page_size', pixel_page_size);
-    else
-        % In contrast to the above case, we can use the loader to get the dnd
-        % as no extra constructor arguments are required.
-        sqw_dnd_obj = ldr.get_dnd(source);
-    end
-    ldr.delete();
+    sqw_dnd_obj=obj_from_faccessor(ldr);
```

Comment:
```suggestion
    sqw_dnd_obj=obj_from_faccessor(ldr);
    ldr.delete();
```

Reply:
no -- it destroys the accessor and closes the file -- the thing you do not want to do as the next operation would be doing filebased cut, i.e. reading pixels. The interface of filebased sqw objects is confusing to me but Tessella introduced it, find it convenient and spent substantial time on development. After some thinking I would get used to this interface.